src/Core/Form/IdentifiableObject/CommandBuilder/CommandField.php

```diff
@@ -28,77 +28,70 @@
 
 namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder;
 
-use Symfony\Component\PropertyAccess\PropertyPath;
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
 
 class CommandField
 {
-    public const TYPE_STRING = 'string';
-    public const TYPE_BOOL = 'bool';
-    public const TYPE_INT = 'int';
-    public const TYPE_ARRAY = 'array';
-    public const TYPE_DATETIME = 'datetime';
-
-    public const ACCEPTED_TYPES = [
-        self::TYPE_STRING,
-        self::TYPE_BOOL,
-        self::TYPE_INT,
-        self::TYPE_ARRAY,
-        self::TYPE_DATETIME,
-    ];
-
-    /**
-     * @var PropertyPath
-     */
-    private $dataPath;
-
     /**
      * @var string
      */
     private $commandSetter;
 
     /**
-     * @var string
+     * @var array<int, DataField>
      */
-    private $type;
+    private $dataFields;
 
     /**
      * @var bool
      */
-    private $multiShopField;
+    private $isMultiShopField;
+
+    protected function __construct(string $commandSetter, array $dataFields, bool $isMultiShopField)
+    {
+        if (empty($dataFields)) {
+            throw new InvalidArgumentException(
+                sprintf('No data field provided to command setter "%s"', $commandSetter)
+            );
+        }
+        foreach ($dataFields as $dataField) {
+            if (!$dataField instanceof DataField) {
+                throw new InvalidArgumentException(sprintf(
+                    'Invalid data field type "%s", expected "%s"',
+                    is_object($dataField) ? get_class($dataField) : gettype($dataField),
+                    DataField::class
+                ));
+            }
+        }
+        $this->commandSetter = $commandSetter;
+        $this->dataFields = array_values($dataFields);
```

Comment:
Why do you need `array_values` here? Is there any case where it could be indexed by keys? Would it even be a problem anyway?

Reply:
Because `$dataFields` is an argument that could have `integer` or `string` keys, and I decided it should have `integer` keys only.